### ISSUE TITLE
Warning about "no backend" even though backend is present (and working)

### ISSUE THREAD

**Issue by edsko** [atom-haskell/ide-haskell#78]
When I have both the `haskell-ghc-mod` _and_ `ide-haskell-cabal` backends present, I get no warning. However, when I disable the `haskell-ghc-mod` package, I get this warning on startup:

> autocomplete-haskell: autocomplete-haskell requires a package providing haskell-completion-backend service. Consider installing haskell-ghc-mod or other package, which provides haskell-completion-backend. You can disable this message in autocomplete-haskell settings.

**Comment by lierdakil** [comment]
`autocomplete-haskell` is a whole other beast. It needs ghc-mod for
browsing modules and whatnot, so it can't work with cabal alone. That's not
something we can do about if there is only cabal backend available, except
disable autocomplete-haskell (or just this message, since with no backend,
autocomplete-haskell simply bails on autocompletion requests, at which
point default autocompletion should take over).

2015-07-04 18:13 GMT+03:00 Edsko de Vries <notifications@github.com>:

**Comment by lierdakil** [comment]
TL;DR, that's not ide-haskell, but a separate package, used exclusively for autocompletion: https://github.com/atom-haskell/autocomplete-haskell

**Comment by edsko** [comment]
Right, I see. Fair enough, I guess I'd have to implement a different services if `cabal` was to support this.

**Comment by lierdakil** [comment]
That won't be easy though, since that'd need to parse Haskell sources (including external modules), which is nigh impossible with pure javascript. Maybe ghcjs could help with that, but that would need much work, I reckon...

**Comment by edsko** [comment]
Oh, yes. For sure. I tend to just use `hasktags` and Atom's built-in tag-based autocompletion (as well as jump to definition).
